server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java

```diff
@@ -107,8 +107,7 @@ private void run() {
         LOGGER.info("Queue={}", queue);
 
         disposable = Flux.from(queue.deQueue())
-            .publishOn(spooler)
-            .flatMap(this::handleOnQueueItem)
+            .flatMap(item -> handleOnQueueItem(item).subscribeOn(spooler))
```

Comment:
what is the difference?

Reply:
The difference is very big actually: without that, the `handleOnQueueItem` will prevent the concurrency to happen because flatMap will run in `immediate` scheduler.

Reply:
I do not get to the same conclusion:

```
@Test
void matthieu() throws InterruptedException {
   toto("direct").apply(0);

   Mono.just(42)
       .flatMap(toto("nothing"))
       .subscribe();

    Mono.just(42)
        .publishOn(Schedulers.boundedElastic())
        .flatMap(toto("publishOn"))
        .subscribe();

    Mono.just(42)
        .flatMap(x -> toto("sub-subscribeOn").apply(x).subscribeOn(Schedulers.boundedElastic()))
        .subscribe();

    Thread.sleep(1000);
}

private <A> Function<A, Mono<A>> toto(String label) {
    return ignored -> Mono.fromRunnable(() -> System.out.println("toto for " + label + " on " + Thread.currentThread().getName())).thenReturn(ignored);
}
```

Gives:

```
toto for nothing on main
toto for publishOn on boundedElastic-1
toto for sub-subscribeOn on boundedElastic-2
```

Reply:
Actually, you get the same conclusion. if you generate several items, you'll see that in the second case, they will all be executed in boundedElastic-1, i.e. `immediate` for the given context (after publishOn scheduler switch)